lib/Test/Pod/CoverageChange.pm

```diff
@@ -0,0 +1,236 @@
+package Test::Pod::CoverageChange;
+# ABSTRACT: check perl modules against their pod coverage
+
+=pod
+
+=head1 NAME
+
+Pod coverage calculator test file.
+
+=head1 SYNOPSIS
+
+It checks all files that placed under a given directory against their
+POD coverage to see if all existing subs have POD and also
+POD syntax to see if is there an POD syntax error or not?
+
+=head1 DESCRIPTION
+
+Prints the percentage of POD coverage in B<TODO> test format for the packages that we allowed to have naked subs.
+Prints an error message if our latest changes increased/decreased numbers of naked sub for the packages that we allowed to have naked sub.
+Prints an error message if a naked allowed package has 100% POD coverage. (We should remove it from the C<%naked_packages> variable list.)
+Ignores to check every package that we pass as C<@ignored_package>
+Prints a proper message for the newly added packages.
+
+Prints C<ok> for the files that have no POD syntax error.
+Prints C<not ok- There is no POD in the file> if the file has no POD at all. I will consider as a pass test since it is in a todo section.
+Prints C<not ok- The number of errors in the POD structure> if the file has any error. It causes tests to be fail.
+
+=cut
+
+use strict;
+use warnings;
+
+use Test::More;
+use Pod::Checker;
+use Pod::Coverage;
+use File::Find::Rule;
+use Test::Pod::Coverage;
+use Module::Path 'module_path';
+
+our $VERSION='0.01';
+
+use constant {
+  POD_SYNTAX_IS_OK => 0,
+  FILE_HAS_NO_POD => -1,
+};
+
+use Exporter qw(import export_to_level);
+our @EXPORT_OK = qw(check);
+
+# Set caller test file name
+my $caller_test_file = (caller())[1];
+
+=pod
+
+=head2 check
+
+Check all modules under a given directory against POD coverage and POD syntax
+
+=over 4
+
+=item * C<directory> - arrayref of directories to check pod against
+
+=item * C<naked_packages> A hashref that contains some packages which are allowed to have naked subs.
+
+=item * C<ignored_packages> An arrayref that contains some packages which we are going to ignore to check.
+
+=back
+
+=cut
+
+sub check {
+    my $path = shift;
+    my $allowed_naked_packages = shift // {};
+    my $ignored_packages = shift // [];
+
+    $path = [$path] unless ref $path eq 'ARRAY';
+    $ignored_packages = [$ignored_packages] unless ref $ignored_packages eq 'ARRAY';
+
+    check_pod_coverage($path, $allowed_naked_packages, $ignored_packages);
+    check_pod_syntax($path, $ignored_packages);
+}
+
+=head2 check_pod_coverage
+
+Checks POD coverage for all the modules that exists under a given directory.
+Passes the $allowed_naked_packages to the L<Test::Pod::CoverageChange::check_allowed_naked_packages>
+Ignores the packages in the C<$ignored_packages> parameter
+
+=over 4
+
+=item C<$directories> Arrayref|String - I works on all the modules that lies under this given directory(ies).
```

Comment:
again, please look at other examples - we don't use this style anywhere else in the code

```suggestion
=item C<$directories> - directories to check recursively, supports string or arrayref
```